owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt

```diff
@@ -99,100 +112,131 @@ class UpdateRemoteShareOperation
+        response: String?,
+        status: Int
+    ): RemoteOperationResult<ShareResponse> {
+        Timber.e("Failed response while while updating remote shares ")
+        if (response != null) {
+            Timber.e("*** status code: $status; response message: $response")
+        } else {
+            Timber.e("*** status code: $status")
+        }
+        return RemoteOperationResult(method)
+    }
 
-            // Parameters to update
-            if (name != null) {
-                formBodyBuilder.add(PARAM_NAME, name!!)
-            }
+    private fun onRequestSuccessful(response: String?): RemoteOperationResult<ShareResponse> {
+        val result = RemoteOperationResult<ShareResponse>(RemoteOperationResult.ResultCode.OK)
+        Timber.d("Successful response: $response")
+        result.data = parseResponse(response!!)
+        Timber.d("*** Retrieve the index of the new share completed ")
+        val emptyShare = result.data.shares.first()
+
+        return if (retrieveShareDetails) {
+            // retrieve more info - PUT only returns the index of the new share
+            GetRemoteShareOperation(emptyShare.id).execute(client)
+        } else {
+            result
+        }
+    }
 
-            if (password != null) {
-                formBodyBuilder.add(PARAM_PASSWORD, password!!)
-            }
+    private fun createFormBodyBuilder(): FormBody.Builder {
+        val formBodyBuilder = FormBody.Builder()
 
-            if (expirationDateInMillis < INITIAL_EXPIRATION_DATE_IN_MILLIS) {
-                // clear expiration date
-                formBodyBuilder.add(PARAM_EXPIRATION_DATE, "")
-
-            } else if (expirationDateInMillis > INITIAL_EXPIRATION_DATE_IN_MILLIS) {
-                // set expiration date
-                val dateFormat = SimpleDateFormat(FORMAT_EXPIRATION_DATE, Locale.getDefault())
-                val expirationDate = Calendar.getInstance()
-                expirationDate.timeInMillis = expirationDateInMillis
-                val formattedExpirationDate = dateFormat.format(expirationDate.time)
-                formBodyBuilder.add(PARAM_EXPIRATION_DATE, formattedExpirationDate)
-            } // else, ignore - no update
-
-            if (publicUpload != null) {
-                formBodyBuilder.add(PARAM_PUBLIC_UPLOAD, publicUpload.toString())
-            }
+        // Parameters to update
+        if (name != null) {
+            formBodyBuilder.add(PARAM_NAME, name!!)
+        }
 
-            // IMPORTANT: permissions parameter needs to be updated after mPublicUpload parameter,
-            // otherwise they would be set always as 1 (READ) in the server when mPublicUpload was updated
-            if (permissions > DEFAULT_PERMISSION) {
-                // set permissions
-                formBodyBuilder.add(PARAM_PERMISSIONS, permissions.toString())
-            }
+        if (password != null) {
+            formBodyBuilder.add(PARAM_PASSWORD, password!!)
+        }
```

Comment:
I know you can do this without that `!!` 😉

Reply:
👍🏻